### ISSUE TITLE
Feature: Custom fetch function for proxies

### ISSUE THREAD

**Issue by OIRNOIR** [amaribot/amaribot.js#23]
I've made a proxy with CloudFlare workers to hide my exact IP address. I want to be able to use it with amaribot.js, but amaribot.js only uses the regular fetch API, which doesn't go through the proxy.

I would like to be able to replace the fetch module within amaribot.js as a config option upon initiating the client with one of my own methods or functions, such as this one:
```js
/**
 * @param {String} endpoint 
 * @param {RequestInit} options 
 * @returns {Promise<Response>}
 */
async function fetch(endpoint, options={}) {
    return new Promise(async (resolve) => {
        const res = await fetch("https://uri.to.my.request.proxy.worker.oirnoir.workers.dev/", {...options, headers: {endpoint, headers: options.headers == null ? "{}" : JSON.stringify(options.headers)}, cache: "no-cache"});
        const receivedHeaders = res.headers.has("incomingHeaders") ? JSON.parse(res.headers.get("incomingHeaders")) : null;
        res.proxyHeaders = res.headers;
        if (receivedHeaders != null) {
            Object.defineProperty(res, "headers", {get: () => new Headers(receivedHeaders)});
        }
        return resolve(res);
    });
}
```
The function has the exact same syntax as the built-in fetch API, but requests through my cloudflare workers proxy. It returns the data in the exact same format. I want to be able to replace the fetch api within amaribot.js with my own function (it can do anything, it just needs to accept and return similar values as the fetch api would). The function above is just an example input.

**Comment by OIRNOIR** [comment]
I would submit a PR but idk how to specify an unknown function as a parameter in ts

**Comment by thewilloftheshadow** [comment]
Implemented in v2.1.0
